mkdocs/docs/HPC/setting_up_python_virtual_environments.md

```diff
@@ -0,0 +1,221 @@
+# Python Virtual Environments (venv's)
+
+## Introduction
+
+Python virtual environments (venv's) are a way to create isolated Python environments.
+You can install packages in a venv without affecting the system Python installation.
+Because a normal user cannot install packages globally, 
+venv's are a way to install packages without needing root access.
+This is useful when you need to use a package that is not available as a module on the HPC cluster.
+
+## Creating a venv
+
+A venv can be created with the following commands:
+
+```bash
+$ python -m venv myenv      # Create a new venv named myenv
+$ source myenv/bin/activate # Activate the venv
+```
+
+After activating the venv, you can install packages with `pip`:
+
+```bash
+$ pip install example_package1
+$ pip install example_package2
+```
+
+It is now possible to run Python scripts that use the installed packages in the venv. To deactivate the venv, run:
+
+```bash
+$ deactivate
+```
+
+!!! note
+    Always prefer to use modules if they are available. 
```

Comment:
You need to clarify what "modules" means here, since there's a lot of room for confusion here with "Python modules".

```suggestion
    Always prefer to use Python packages that are centrally installed, which are available via the environment modules interface.
```

Also link to the section on environment modules we have already.